app/Http/Controllers/Api/V1/SearchController.php

```diff
@@ -0,0 +1,224 @@
+     * 
+     * @response  {
+     *     "projects":{
+     *        "457":{
+     *           "id":457,
+     *           "name":"Text Structure Lesson 2 Problem and Solution",
+     *           "description":"Learning Objective\nThe learner will define and describe the main characteristics...",
+     *           "thumb_url":"/storage/uploads/3zjZuLoQrRk0MZ5wP7UPyOut5zUybf3tW3a4Q2M1.png",
+     *           "mongo_userid":"5ef5300e41668b53ea5ed1b3",
+     *           "starter_project":false,
+     *           "created_at":"2020-07-17T17:49:01.000000Z",
+     *           "updated_at":"2020-08-06T13:28:07.000000Z",
+     *           "deleted_at":null,
+     *           "playlists":{
+     *              "225":{
+     *                 "id":225,
+     *                 "title":"Solving Ratio and Rate Problems",
+     *                 "project_id":64,
+     *                 "order":null,
+     *                 "mongo_projectid":"5f3ae92a924a1d5ddf44dd80",
+     *                 "created_at":null,
+     *                 "updated_at":null,
+     *                 "deleted_at":null,
+     *                 "activities":{
+     *                    "993":{
+     *                       "id":993,
+     *                       "playlist_id":225,
+     *                       "title":"",
+     *                       "type":"h5p",
+     *                       "content":"",
+     *                       "h5p_content_id":17474,
+     *                       "thumb_url":"/storage/uploads/5f3aedeba24fb.jpeg",
+     *                       "subject_id":null,
+     *                       "education_level_id":null,
+     *                       "shared":false,
+     *                       "order":11,
+     *                       "mongo_playlistid":"5f3aeccd924a1d5ddf44ddd0",
+     *                       "created_at":null,
+     *                       "updated_at":null,
+     *                       "deleted_at":null
+     *                    }
+     *                 }
+     *              }
+     *           }
+     *        }
+     *     }
+     * }
+     * @response  400 {
+     *     "errors": {
+     *         "query": [
+     *             "The query field is required."
+     *         ]
+     *     }
+     * }
+     * @param  Request  $request
+     * @return Response
+     */
+    public function search(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+            'query' => 'required|string|max:255',
+            'sort' => 'in:created_at',
+            'order' => 'in:asc,desc',
+            'from' => 'integer',
+            'size' => 'integer'
+        ]);
+
+        if ($validator->fails()) {
+            return response([
+                'errors' => $validator->errors()
+            ], 400);
+        }
+
+        $projects = $this->activityRepository->searchForm($request);
+
+        return response([
+            'projects' => $projects,
+        ], 200);
```

Comment:
Please use Resource for response formatting.

Reply:
Please suggest how to use Resource for response formatting in this case ?
In this case we are searching keyword from three different tables projects, playlists and activities and then we have generate response showed in the attached screen shot.
So say we match an activity then we need to get its playlist, project and get them in nested format.
Similarly if match playlist, we need to check if its playlist is in nested format or not, if not not then get it etc

![image](https://user-images.githubusercontent.com/63652626/91977146-9eb90380-ed3b-11ea-86af-cfdee55c83ed.png)

Reply:
@yurysakovich1010  - this is something we'll need to discuss with @fahad-curriki . 
I was thinking, that maybe, we need separate resources for each of the collections he needs. We already have resources for projects, activities and playlists, no?

Reply:
@aqeel-tkxel @fahad-curriki We already have resources for Activity, Playlist and Project.
We can use combination of that. If you need, you can add new Resource.

Reply:
@yurysakovich1010 I can use Resource classes for individual Activity, Playlist and Project,
But in the end the response will be an array not a Resource,
So should I return the array of nested Resources for Activity, Playlist and Project ?

Reply:
PlaylistResource::collection

Reply:
@yurysakovich1010 Will passing array of Resource Responses to "::collection" work ?
I think "::collection" expects Model Collection.

Reply:
@aqeel-tkxel Added Resource response for individual Activity, Playlist and Project.